tests/test_gruptree.py

```diff
@@ -123,6 +139,54 @@ def test_grupnetdf():
     assert 100 in grupdf["TERMINAL_PRESSURE"].values
 
 
+@pytest.mark.parametrize(
+    "dicttree, expected_str",
+    [
+        ({}, ""),
+        ({"foo": {}}, "foo"),
+        ({"foo": {"bar": {}}}, "foo\n└── bar"),
+        ({"foo": {"bar": {}, "com": {}}}, "foo\n├── bar\n└── com"),
+        # Test sorting:
+        ({"foo": {"com": {}, "bar": {}}}, "foo\n├── bar\n└── com"),
+        # Two levels:
+        (
+            {"foo": {"bar": {}, "com": {"fjooo": {}}}},
+            """
+foo
+├── bar
+└── com
+    └── fjooo""",
+        ),
+        # Integers as node names:
+        ({1: {2: {}}}, "1\n└── 2"),
+        # More complex structure:
+        # Note: Node names cannot be duplicated, even on
+        # unique branches:
```

Comment:
Where does this restriction stem from?

Reply:
It is from how `treelib` is used. I think the restriction can be relaxed within `ecl2df`, but have not tested.

Reply:
Would it invalidate data that otherwise is valid from the users perspective (as we are looking at representations of `GRUPNET` structures).

Reply:
`GRUPTREE` (standard network model) I am quite sure do not allow repeated nodes (will give loops and recycled production(?)). Not that certain for `GRUPNET` (extended network model), the manual states that the tree must have "gathering tree topology", whatevery they mean by that. Extended network model is not yet supported by ecl2df.